### ISSUE TITLE
Unable to recognize composite verb with some (but not all) kanji written in kana

### ISSUE THREAD

**Issue by SaltfishAmi** [birtles/rikaichamp#302]
つれこむ and 連れ込む can be recognized as one verb, but either 連れこむ or つれ込む could not.

This behavior seems reasonable as we don't want it to read 沢さん as 沢山, but for verbs especially simple composite verb components like 込む in 連れこむ,  抜ける in 駆けぬける, there is a good chance a Japanese would write them in kana.

On the other hand, 使い切る and 使いきる can both be recognized correctly. Maybe there're both entries in the dictionary?

**Comment by birtles** [comment]
Yes, that's right. Many verbs have both entries in the dictionary. e.g. 片づける and 片付ける.

It would be pretty tricky to recognize all the variations automatically. Handling just the first case, 連れこむ, might be possible with some pretty involved rules such as:

1. One of the longest matches is a masu-stem and the next (unmatched) character is hiragana
2. There is a longer entry in the database with the same masu-stem followed by kanji
3. When converting the database entry to hiragana, the next character(s) match the unmatched hiragana for one of the generated deinflections

It's pretty involved and whether or not a composite verb ends up in the dictionary is already quite arbitrary so I'm not sure how important this would be.

**Comment by nicolasmaia** [comment]
Whenever I stumble upon this problem, I tend to propose the new forms at JMdict instead.

In this case, 連れこむ has [reasonable ngrams (645)](http://nlp.cis.unimelb.edu.au/jwb/ngrams/ngramlookup.cgi?sent=%E9%80%A3%E3%82%8C%E3%81%93%E3%82%80), but つれ込む has [only 38](http://nlp.cis.unimelb.edu.au/jwb/ngrams/ngramlookup.cgi?sent=%E3%81%A4%E3%82%8C%E8%BE%BC%E3%82%80), so I'm going to propose adding the first one.
